includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -376,185 +376,220 @@ public function payment_fields() {
 	/**
 	 * Process the payment for a given order.
 	 *
-	 * @param int $order_id Order ID to process the payment for.
+	 * @param int  $order_id Order ID to process the payment for.
+	 * @param bool $is_recurring_payment Whether this is a one-off payment (false) or it's the first installment of a recurring payment (true).
 	 *
 	 * @return array|null An array with result of payment and redirect URL, or nothing.
 	 */
-	public function process_payment( $order_id ) {
-		$order = wc_get_order( $order_id );
-		return $this->process_payment_for_order( $order, WC()->cart );
+	public function process_payment( $order_id, $is_recurring_payment = false ) {
+		try {
+			// phpcs:ignore WordPress.Security.NonceVerification.Missing
+			$payment_information = WC_Payment_Information::from_payment_request( $_POST, $is_recurring_payment );
+			$order               = wc_get_order( $order_id );
+			$manual_capture      = 'yes' === $this->get_option( 'manual_capture' );
```

Comment:
I don't think we should include the `$manual_capture` variable because that's a plugin setting, and isn't strictly related to the payment information. It's also an additional parameter to the constructor, although that's less important.

I don't feel too strongly about this though. The only reason I haven't included it so far is that this is a plugin setting, not payment information. I'm more than willing to change this if there are more people on the team that think we should include it 🙂 .

The `$order` however could probably be added there. Or rather, the `$order_id` could be passed to the constructor and the order retrieved when the `Payment_Information` class is instantiated.

I'm still a little hesitant to add the `$order` because I feel like that might be its own thing. That feels more like payment _metadata_ rather than payment _information_. It certainly relates to the payment, but it's not _tied_ to the payment.

I'd love input from more people in @Automattic/hydra on this.

Reply:
In that case I'll see if it makes sense to fit this into the `Payment_Information` class along with the `$order_id`.

I'm not sure what that would look like right now, since @RadoslavGeorgiev, @luizreis, and I just made pretty extensive changes—including removing some redundant variables—so I'll take a closer look tomorrow 🙂.

Reply:
Moved this to new issues: #862 and #863 . Marking this discussion as resolved so we can move on with this PR.